### PR TITLE
add provider

### PR DESCRIPTION
**Pull request by rex-9**
- First of all - I made sure that I keep the state immutable in reducers!
- Secondly, changed the default state in the books reducer from an empty array to an array with a few books. One of the goals is to display them in the React components.
- Imported Redux `Provider` and the store in the main component of the React app.
- Connected the app to the Redux store.
- Made sure that list of books displays books from the Redux store.
  - Use `useSelector` to consume the state.
- In the React component responsible for adding new books, add functionality for the "Add new" button click:
  - Imported Redux `useDispatch`.
  - Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
  - Takes the inputs from the state and generates a unique id and adds them to an object.
  - Once the new book object is ready to be submitted to the Redux store, dispatch a corresponding action (imported Action Creator that I created before and use it here).
- In the React component responsible for removing books, add functionality for the "Remove" button click:
  - Import Redux `useDispatch`.
  - Dispatch a corresponding action (import Action Creator that I created before and used it here).
- In the React component responsible for categories, add functionality for the "Check status" button click:
  - Import Redux `useDispatch`.
  - Dispatch a corresponding action (import Action Creator that I created before and use it here).

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -18,14 +19,12 @@ const AddBook = (props) => {
   };
 
   const clickAdd = () => {
-    const { listenChanges } = props;
     if (title === '' || author === '') {
       console.log('form is empty');
```
Comment (Ismailco) [inline src/components/AddBook.js]: - [ ] For best practice, kindly remove all `conslole.log()` it makes your code slower

File: src/components/Categories.js [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+  useEffect(() => {
+    dispatch(checkStatus());
+  }, []);
+  const categories = useSelector((state) => state.categories);
+
   const clickCheckStatus = () => {
-    store.dispatch(checkStatus());
-    console.log(store.getState().categories);
+    dispatch(checkStatus());
+    console.log(categories);
```
Comment (Ismailco) [inline src/components/Categories.js]: - [ ] For best practice, kindly remove all `conslole.log()` it makes your code slower

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @HtetNaing0814, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js (hunk 2 above)
Comment (rex-9) [reply src/components/Categories.js]: I don't like this review. console.log is used like the boilerplate for the coming-up sections of the projects. But I'll remove it temporarily to get approved.

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @HtetNaing0814 

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js (hunk 2 above)
Comment (Ismailco) [reply src/components/Categories.js]: @HtetNaing0814, you should not commit `console.log()` to your Github repo,  there is no reason to use `console.log()` for anything in this project other than debugging

**Pull request merged**